doc/user_manual/existing_interfaces.tex

```diff
@@ -103,6 +103,174 @@ \subsection{Generic Interface}
+\xmlAttr{subType}.  \xmlAttr{name} identifies that particular \xmlNode{Code} model within RAVEN, and
+\xmlAttr{subType} specifies which code interface the model will use (In this case \xmlAttr{subType}=``RAVEN''). 
+The \xmlNode{executable}
+block should contain the absolute or relative (with respect to the current working
+directory) path to the RAVEN framework script (\textbf{raven\_framework}).
+\\ In addition to the attributes and xml nodes reported above, the RAVEN accepts the following XML nodes (required and optional):
+\begin{itemize}
+  \item  \xmlNode{outputExportOutStreams}, \xmlDesc{comma separated list,
+  required parameter} will specify the  \xmlNode{OutStreams} that will be loaded as outputs of the SLAVE RAVEN.
+  Maximum two  \xmlNode{OutStreams} can be listed here (1 for PointSet and/or 1 for HistorySet).
+  \item  \xmlNode{conversionModule}, \xmlDesc{string,
+  optional parameter} will specify the path to a \textit{Python} module that can contain two methods:
+      \begin{itemize}
+         \item \textbf{\textit{manipulateScalarSampledVariables}}, a method that is aimed to manipulate sampled variables and to create more in case needed.
+         Example:
+          \begin{lstlisting}[language=python]
+def manipulateScalarSampledVariables(sampledVariables):
+  """
+  This method is aimed to manipulate scalar variables. 
+  The user can create new variables based on the 
+  variables sampled by RAVEN
+   @ In, sampledVariables, dict, dictionary of 
+       sampled variables ({"var1":value1,"var2":value2})
+   @ Out, None, the new variables should be 
+                 added in the "sampledVariables" dictionary
+  """
+  newVariableValue = 
+    sampledVariables['Distributions|Uniform@name:a_dist|lowerBound'] 
+    + 1.0
+  sampledVariables['Distributions|Uniform@name:a_dist|upperBound'] = 
+    newVariableValue
+  return
+           \end{lstlisting}
+
+         \item \textbf{\textit{convertNotScalarSampledVariables}}, a method that is aimed to convert not scalar variables (e.g. 1D arrays) into multiple scalar variables 
+         (e.g.  \xmlNode{constant}(s) in a sampling strategy).
+          This method is going to be required in case not scalar variables are detected by the interface. 
+          Example:
+          \begin{lstlisting}[language=python]
+ def convertNotScalarSampledVariables(noScalarVariables):
+  """
+  This method is aimed to convert not scalar 
+  variables into multiple scalar variables. The user MUST
+   create new variables based on the not Scalar Variables
+    sampled (and passed in) by RAVEN
+  @ In, noScalarVariables, dict, dictionary of sampled 
+       variables that are not scalar ({"var1":1Darray1,"var2":1Darray2})
+  @ Out, newVars, dict,  the new variables that have 
+       been created based on the not scalar variables 
+       contained in "noScalarVariables" dictionary
+  """
+  oneDimensionalArray =
+      sampledVariables['temperatureHistory']
```

Comment:
should not  sampledVariables be noScalarVariables?

Reply:
addressed